src/reanimated2/hook/useAnimatedStyle.ts

```diff
@@ -411,16 +411,16 @@ export function useAnimatedStyle<Style extends DefaultStyle>(
   isAnimatedProps = false
 ) {
   const viewsRef: ViewRefSet<unknown> = useViewRefSet();
```

Comment:
1. Do we really want to call `useViewRefSet()` on each render?
2. How about calling `useViewRefSet`  conditionally only on web?
```suggestion
  const maybeViewsRef = SHOULD_BE_USE_WEB ? useViewRefSet() : undefined;
```

Reply:
I was considering doing that initially but then scrapped the idea do make this PR touch only `makeRemote`. I can add it here however.

Reply:
Regarding 1. `useViewRefSet` uses `useRef` inside so there's no problem calling it on every render.